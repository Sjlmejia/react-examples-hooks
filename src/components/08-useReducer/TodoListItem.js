import React from 'react'

export const TodoListItem = ({description, id, done, index, handleToggle, handleDelete}) => {
  return (
    <li key={id} className="list-group-item">
      <div
        className={`item ${done && 'complete'}`}
        onClick={() => { handleToggle(id)}}
      >
        {index +1}. {description}
      </div>
      <button className="btn btn-outline-danger" onClick={()=> {handleDelete(id)}}>Borrar</button>
    </li>
  )
}
