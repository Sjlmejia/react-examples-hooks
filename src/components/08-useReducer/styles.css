body {
  padding: 50px;
}
.list-group-item {
  display: flex;
  justify-content: space-between;
}

.item {
  cursor: pointer;
  display: flex;
  align-items: center;
}

.complete {
  text-decoration: line-through;
}