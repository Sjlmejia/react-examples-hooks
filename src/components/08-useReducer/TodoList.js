import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({todos,handleDelete, handleToggle }) => {
  return (
    <ul className="list-group list-group-flush">
        {
          todos.map(({desc, id, done}, i) => {
            return <TodoListItem
                      key={id}
                      description={desc}
                      id={id} done={done}
                      index={i}
                      handleToggle={handleToggle}
                      handleDelete={handleDelete}/>
          })
        }
    </ul>
  )
}
