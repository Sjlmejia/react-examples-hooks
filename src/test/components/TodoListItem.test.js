import { shallow } from "enzyme";
import { TodoListItem } from "../../components/08-useReducer/TodoListItem";

describe('Evaluar Tod List', () => {
  const item = {
    id:1,
    desc: 'Aprender React',
    done: false
  };
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem 
      description={item.desc} id={item.id} done={item.done} index={1} handleToggle={handleToggle} handleDelete={handleDelete}
    />
  )
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe llamar la funcion borrar', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(1);
  });

  test('debe llamar la funcion handleToggle', () => {
    wrapper.find('div').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(1);
  });

  
  test('debe lde mostarse el texto correctamente', () => {
    const text = wrapper.find('div').text();
    expect(text).toEqual('2. Aprender React')
  });

  
  test('debe tener la clase complete si el TODO.done=true', () => {
    const todo = {
      id:1,
      desc: 'Aprender React',
      done: true
    };
    const wrapper = shallow(
      <TodoListItem 
        description={todo.desc} id={todo.id} done={todo.done} index={1}
        handleToggle={handleToggle} handleDelete={handleDelete}
      />
    )
    expect(wrapper.find('div').hasClass('complete')).toBe(true)
  });
})