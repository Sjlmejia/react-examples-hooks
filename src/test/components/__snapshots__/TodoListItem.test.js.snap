// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluar Tod List debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <div
    className="item false"
    onClick={[Function]}
  >
    2
    . 
    Aprender React
  </div>
  <button
    className="btn btn-outline-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
